### ISSUE TITLE
Issue during Training

### ISSUE THREAD

**Issue by nathand2020** [HasnainRaz/Fast-SRGAN#9]
Hi, I used your latest code and trained the model with Div2K.
However, the results are not as good. I noticed that the losses keep fluctuating.
I used the command as follows
python main.py --image_dir ./DIV2K_train_HR --hr_size 384 --lr 1e-4 --save_iter 200 --epochs 30 --batch_size 14
These are the log file after 30 epochs.
[EP1] Adversarial Loss=0.080617	Content Loss=0.001997	MSE Loss=0.026345	Discriminator Loss=0.781968
[EP2] Adversarial Loss=0.132610	Content Loss=0.001199	MSE Loss=0.035348	Discriminator Loss=0.905672
[EP3] Adversarial Loss=0.123783	Content Loss=0.001260	MSE Loss=0.022111	Discriminator Loss=0.938514
[EP4] Adversarial Loss=0.097441	Content Loss=0.001445	MSE Loss=0.015596	Discriminator Loss=1.121814
[EP5] Adversarial Loss=0.122653	Content Loss=0.001467	MSE Loss=0.019071	Discriminator Loss=0.786843
[EP6] Adversarial Loss=0.092276	Content Loss=0.001092	MSE Loss=0.023201	Discriminator Loss=0.901218
[EP7] Adversarial Loss=0.102440	Content Loss=0.001587	MSE Loss=0.018378	Discriminator Loss=0.702091
[EP8] Adversarial Loss=0.110358	Content Loss=0.001532	MSE Loss=0.026881	Discriminator Loss=0.590455
[EP9] Adversarial Loss=0.106875	Content Loss=0.003184	MSE Loss=0.020608	Discriminator Loss=0.387678
[EP10] Adversarial Loss=0.084382	Content Loss=0.002820	MSE Loss=0.021133	Discriminator Loss=0.382028
[EP11] Adversarial Loss=0.077473	Content Loss=0.004879	MSE Loss=0.015751	Discriminator Loss=0.420842
[EP12] Adversarial Loss=0.059834	Content Loss=0.004527	MSE Loss=0.013836	Discriminator Loss=0.307745
[EP13] Adversarial Loss=0.106401	Content Loss=0.006681	MSE Loss=0.019882	Discriminator Loss=0.106354
[EP14] Adversarial Loss=0.084455	Content Loss=0.006933	MSE Loss=0.021794	Discriminator Loss=0.038181
[EP15] Adversarial Loss=0.146453	Content Loss=0.007463	MSE Loss=0.028703	Discriminator Loss=0.057300
[EP16] Adversarial Loss=0.108233	Content Loss=0.008538	MSE Loss=0.021102	Discriminator Loss=0.021846
[EP17] Adversarial Loss=0.096775	Content Loss=0.010486	MSE Loss=0.016221	Discriminator Loss=0.137891
[EP18] Adversarial Loss=0.107159	Content Loss=0.007517	MSE Loss=0.021911	Discriminator Loss=0.029221
[EP19] Adversarial Loss=0.088703	Content Loss=0.008759	MSE Loss=0.009813	Discriminator Loss=0.192226
[EP20] Adversarial Loss=0.077869	Content Loss=0.010124	MSE Loss=0.018487	Discriminator Loss=0.073865
[EP21] Adversarial Loss=0.108033	Content Loss=0.008940	MSE Loss=0.020208	Discriminator Loss=0.020908
[EP22] Adversarial Loss=0.059458	Content Loss=0.007813	MSE Loss=0.018918	Discriminator Loss=0.023339
[EP23] Adversarial Loss=0.089555	Content Loss=0.004957	MSE Loss=0.022235	Discriminator Loss=0.076594
[EP24] Adversarial Loss=0.128833	Content Loss=0.009217	MSE Loss=0.023711	Discriminator Loss=0.017186
[EP25] Adversarial Loss=0.092714	Content Loss=0.007943	MSE Loss=0.028336	Discriminator Loss=0.020029
[EP26] Adversarial Loss=0.110232	Content Loss=0.007033	MSE Loss=0.019300	Discriminator Loss=0.024588
[EP27] Adversarial Loss=0.081943	Content Loss=0.006822	MSE Loss=0.022644	Discriminator Loss=0.024508
[EP28] Adversarial Loss=0.109066	Content Loss=0.008239	MSE Loss=0.027397	Discriminator Loss=0.017921
[EP29] Adversarial Loss=0.096089	Content Loss=0.007800	MSE Loss=0.028556	Discriminator Loss=0.009476
[EP30] Adversarial Loss=0.093064	Content Loss=0.008842	MSE Loss=0.019332	Discriminator Loss=0.023928

This is an example of training image after 30 epochs.
Generated Image
![generated](https://user-images.githubusercontent.com/63071956/78293493-3ea10580-74ee-11ea-8a5a-58739f9ca7f5.png)
HR image:
![HR](https://user-images.githubusercontent.com/63071956/78293499-3fd23280-74ee-11ea-882d-91155f2ae823.png)
LR image:
![LR](https://user-images.githubusercontent.com/63071956/78293501-41035f80-74ee-11ea-8267-5962674da386.png)

I saw the results are not very different from bilinear upsampling and still far from your provided pretrained model. Do you have any recommendation to fix this issue? 
Thanks

**Comment by HasnainRaz** [comment]
If you are training on div2k, it takes a lot more than 30 epochs to train this network. The provided pretrained model was trained for 2 days.

You may also want to try and rebalance the losses (change their weighting). As far as the losses are concerned, the will bounce around alot as it's a minimax game.

Lastly, I think the best you can hope for is what the provided pretrained model does, as this is a compact and fast model, and as a result of this constraint, will not be able to beat the SOTA SR models.

**Comment by nathand2020** [comment]
Thank Hasnain for your reply. I do agree that the fast version cannot beat the SOTA SR models. 
I just want to ask for the config to get the results as your pretrained model. 
Do you have any advice about the starting point of weighting the losses to get the results as your pretrained model? That would be more than enough for me, :).

**Comment by HasnainRaz** [comment]
@nathand2020, I think the default config in this repo is the one I used, I think your main problem is train time. Set the number of epochs to a very high value (~100k) and observe if the training results improve over the course of the day. If they do, let it run for another day.

**Comment by nathand2020** [comment]
@HasnainRaz : Thank you for your information, I will try with more epochs as your suggestion.
